shell.go

```diff
@@ -20,11 +22,42 @@ import (
 	tar "github.com/whyrusleeping/tar-utils"
 )
 
+const (
+	DefaultPathName = ".ipfs"
+	DefaultApiFile  = "api"
+	EnvDir          = "IPFS_PATH"
+)
+
 type Shell struct {
 	url     string
 	httpcli *gohttp.Client
 }
 
+func NewLocalShell() *Shell {
+	baseDir := os.Getenv(EnvDir)
+	if baseDir == "" {
+		u, err := user.Current()
+		if err != nil {
+			return nil
+		}
+
+		baseDir = path.Join(u.HomeDir, DefaultPathName)
+	}
+
+	apiFile := path.Join(baseDir, DefaultApiFile)
+
+	if _, err := os.Stat(apiFile); err != nil {
+		return nil
+	}
+
+	api, err := ioutil.ReadFile(apiFile)
```

Comment:
Maybe better to strip whitespace from the end of the file content. Lots of editors add a trailing newline, and `echo` does too.